src/Microsoft.VisualStudio.LanguageServices.Razor/TagHelperDescriptorJsonConverter.cs

```diff
@@ -0,0 +1,234 @@
+                .NameComparisonMode((RequiredAttributeDescriptor.NameComparisonMode)nameComparison)
+                .Value(value)
+                .ValueComparisonMode((RequiredAttributeDescriptor.ValueComparisonMode)valueComparison);
+
+            foreach (var diagnostic in diagnostics)
+            {
+                builder.AddDiagnostic(ReadDiagnostic(diagnostic.Value<JObject>(), serializer));
+            }
+        }
+
+        private void ReadBoundAttribute(ITagHelperBoundAttributeDescriptorBuilder builder, JObject attribute, JsonSerializer serializer)
+        {
+            var descriptorKind = attribute[nameof(BoundAttributeDescriptor.Kind)].Value<string>();
+            if (descriptorKind != ITagHelperBoundAttributeDescriptorBuilder.DescriptorKind)
+            {
+                throw new NotSupportedException();
+            }
+
+            var name = attribute[nameof(BoundAttributeDescriptor.Name)].Value<string>();
+            var typeName = attribute[nameof(BoundAttributeDescriptor.TypeName)].Value<string>();
+            var isEnum = attribute[nameof(BoundAttributeDescriptor.IsEnum)].Value<bool>();
+            var indexerNamePrefix = attribute[nameof(BoundAttributeDescriptor.IndexerNamePrefix)].Value<string>();
+            var indexerTypeName = attribute[nameof(BoundAttributeDescriptor.IndexerTypeName)].Value<string>();
+            var documentation = attribute[nameof(BoundAttributeDescriptor.Documentation)].Value<string>();
+            var diagnostics = attribute[nameof(BoundAttributeDescriptor.Diagnostics)].Value<JArray>();
+            var metadata = attribute[nameof(BoundAttributeDescriptor.Metadata)].Value<JObject>();
+
+            builder
+                .Name(name)
+                .TypeName(typeName)
+                .Documentation(documentation)
+                .AsDictionary(indexerNamePrefix, indexerTypeName);
+
+            if (isEnum)
+            {
+                builder.AsEnum();
+            }
+
+            foreach (var diagnostic in diagnostics)
+            {
+                builder.AddDiagnostic(ReadDiagnostic(diagnostic.Value<JObject>(), serializer));
+            }
+
+            var metadataValue = serializer.Deserialize<Dictionary<string, string>>(metadata.CreateReader());
+            foreach (var item in metadataValue)
+            {
+                if (string.Equals(item.Key, ITagHelperBoundAttributeDescriptorBuilder.PropertyNameKey, StringComparison.Ordinal))
+                {
+                    builder.PropertyName(item.Value);
+                }
+
+                builder.AddMetadata(item.Key, item.Value);
+            }
+        }
+
+        private RazorDiagnostic ReadDiagnostic(JObject diagnostic, JsonSerializer serializer)
+        {
+            var span = diagnostic[nameof(RazorDiagnostic.Span)].Value<JObject>();
+            var absoluteIndex = span[nameof(SourceSpan.AbsoluteIndex)].Value<int>();
+            var lineIndex = span[nameof(SourceSpan.LineIndex)].Value<int>();
```

Comment:
Not quite. While deserializing `SourceSpan`, the default deserializer tries to use the constructor with `SourceLocation` and just ends up initializing it with default values.